rakelib/utils.rake

```diff
@@ -0,0 +1,58 @@
+# run a command, pipe output through 'xcpretty' and store the output in CI artifacts
+def xcpretty(cmd, task, subtask = '')
+  name = (task.name + (subtask.empty? ? '' : "_#{subtask}")).gsub(/[:-]/, "_")
+  command = [*cmd].join(' && ')
+  xcpretty = `which xcpretty`
+
+  if ENV['CI']
+    sh "set -o pipefail && (#{command}) | tee \"#{ENV['CIRCLE_ARTIFACTS']}/#{name}_raw.log\" | xcpretty --color --report junit --output \"#{ENV['CIRCLE_TEST_REPORTS']}/xcode/#{name}.xml\""
+  elsif xcpretty && $?.success?
+    sh "set -o pipefail && (#{command}) | xcpretty -c"
+  else
+    sh command
+  end
+end
+
+# run a command and store the output in CI artifacts
+def plain(cmd, task, subtask = '')
```

Comment:
Yes to keeping the other methods separate and make them private and just make the main method private. 

But I'm not entirely convinced in detecting the command and choosing of we should `xcpretty` or `xcrun` or nothing based on checking if the command contains some key words like `xcodebuild`, would make too much intelligence in the utility method imho and not really respecting responsibility principle. That's why I'd prefer the call site to decide that.